projects/organization-management/src/app/store/cost-centers/cost-centers.effects.ts

```diff
@@ -0,0 +1,194 @@
+                  messageParams: { 0: `${costCenter.name}` },
+                }),
+              ])
+            )
+          ),
+          mapErrorToAction(addCostCenterFail)
+        )
+      )
+    )
+  );
+
+  updateCostCenter$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(updateCostCenter),
+      mapToPayloadProperty('costCenter'),
+      concatMap(cc =>
+        this.costCentersService.updateCostCenter(cc).pipe(
+          concatMap(costCenter =>
+            this.navigateTo('../').pipe(
+              mergeMapTo([
+                updateCostCenterSuccess({ costCenter }),
+                displaySuccessMessage({
+                  message: 'account.organization.cost_center_management.update.confirmation',
+                  messageParams: { 0: `${costCenter.name}` },
+                }),
+              ])
+            )
+          ),
+          mapErrorToAction(updateCostCenterFail)
+        )
+      )
+    )
+  );
+
+  deleteCostCenter$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(deleteCostCenter),
+      mapToPayloadProperty('id'),
+      exhaustMap(id =>
+        this.costCentersService.deleteCostCenter(id).pipe(
+          mergeMap(() => [
+            deleteCostCenterSuccess({ id }),
+            displaySuccessMessage({
+              message: 'account.organization.cost_center_management.delete.confirmation',
+            }),
+          ]),
+          mapErrorToAction(deleteCostCenterFail)
+        )
+      )
+    )
+  );
+
+  addCostCenterBuyers$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(addCostCenterBuyers),
+      mapToPayload(),
+      mergeMap(payload =>
+        this.costCentersService.addCostCenterBuyers(payload.costCenterId, payload.buyers).pipe(
+          concatMap(costCenter => this.navigateTo('../').pipe(map(() => addCostCenterBuyersSuccess({ costCenter })))),
+          mapErrorToAction(addCostCenterBuyersFail)
+        )
+      )
+    )
+  );
+
+  updateCostCenterBuyer$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(updateCostCenterBuyer),
+      mapToPayload(),
+      mergeMap(payload =>
+        this.costCentersService.updateCostCenterBuyer(payload.costCenterId, payload.buyer).pipe(
+          map(costCenter => updateCostCenterBuyerSuccess({ costCenter })),
+          mapErrorToAction(updateCostCenterBuyerFail)
+        )
+      )
+    )
+  );
+
+  deleteCostCenterBuyer$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(deleteCostCenterBuyer),
+      mapToPayload(),
+      mergeMap(payload =>
+        this.costCentersService.deleteCostCenterBuyer(payload.costCenterId, payload.login).pipe(
+          map(costCenter => deleteCostCenterBuyerSuccess({ costCenter })),
+          mapErrorToAction(deleteCostCenterBuyerFail)
+        )
+      )
+    )
+  );
+
+  private navigateTo(path: string) {
+    // find current ActivatedRoute by following first activated children
+    let currentRoute = this.router.routerState.root;
```

Comment:
I guess, this code is necessary because it is part of a (standalone) project. I copied this method from the user management. (authored by Danilo and Francisco)